float64 p_traj
